editoast/src/main.rs

```diff
@@ -717,6 +747,92 @@ mod tests {
         tmp_file
     }
 
+    #[actix_test]
+    async fn clone_infra_returns_new_cloned_infra() {
+        // GIVEN
+        let pg_config = PostgresConfig::default();
+        let pg_config_url = pg_config.url().expect("cannot get postgres config url");
+        let manager = ConnectionManager::<PgConnection>::new(pg_config_url.as_str());
+        let db_pool = crate::Data::new(crate::Pool::builder(manager).build().unwrap());
+
+        let infra = Infra {
+            name: Some("test".into()),
+            created: Some(Utc::now().naive_utc()),
+            owner: Some(Uuid::nil()),
+            railjson_version: Some("3.4.6".into()),
+            version: Some("0".into()),
+            generated_version: Some(Some("0".into())),
+            locked: Some(false),
+            ..Default::default()
+        };
+
+        let created_infra = infra
+            .create(db_pool)
+            .await
+            .expect("cannot create a new infra");
+
+        let infra_args: InfraCloneArgs = InfraCloneArgs {
+            id: created_infra.id.unwrap(),
+            new_name: "cloned_infra_1".to_string(),
+        };
+
+        // WHEN
+        let result = clone_infra(infra_args.clone(), pg_config.clone()).await;
+
+        // THEN
+        assert!(result.is_ok());
+        assert_eq!(result.unwrap().name.unwrap(), infra_args.new_name);
+
+        // CLEANUP
+        let pg_config_url = pg_config.url().expect("cannot get postgres config url");
+        let mut conn = PgConnection::establish(pg_config_url.as_str())
+            .await
+            .expect("Error while connecting DB");
+        sql_query("DELETE FROM infra WHERE name = $1")
+            .bind::<Text, _>(infra_args.new_name)
+            .execute(&mut conn)
+            .await
+            .unwrap();
+    }
+
+    #[actix_test]
+    async fn clone_infra_returns_infra_not_found() {
+        // GIVEN
+        let pg_config = PostgresConfig::default();
+        let infra_args: InfraCloneArgs = InfraCloneArgs {
+            id: 1234567890,
+            new_name: "cloned_infra_1".to_string(),
+        };
+
+        // WHEN
+        let result = clone_infra(infra_args.clone(), pg_config.clone()).await;
+
+        // THEN
+        assert!(result.is_err());
+        let error = result.unwrap_err();
+        assert_eq!(error.error_type, "editoast:infra:NotFound");
+        assert_eq!(error.message, "Infrastructure not found, ID: 1234567890");
+    }
+
+    #[actix_test]
+    async fn clone_infra_returns_error() {
```

Comment:
It's not up to editoast to check that the Postgres connection has the appropriate behavior when its settings are wrong. I'd say you can remove this test.

Reply:
Thanks for reviewing! I included the Postgres connection test as an example to highlight a specific error we need to handle differently in our infrastructure (infra not found). Other errors are forwarded to the main.